utils/generate_types/overrides-test.d.ts

```diff
@@ -341,38 +341,92 @@ type AllMatchers<R, T> = PageAssertions & LocatorAssertions & APIResponseAsserti
+  pluralize(word: string, count: number): string;
+  printDiffOrStringify(expected: unknown, received: unknown, expectedLabel: string, receivedLabel: string, expand: boolean): string;
+  printExpected(value: unknown): string;
+  printReceived(object: unknown): string;
+  printWithType<T>(name: string, value: T, print: (value: T) => string): string;
+  replaceMatchedToAsymmetricMatcher(replacedExpected: unknown, replacedReceived: unknown, expectedCycles: Array<unknown>, receivedCycles: Array<unknown>): { replacedExpected: unknown; replacedReceived: unknown };
+  stringify(object: unknown, maxDepth?: number, maxWidth?: number): string;
+}
+
+type State = {
+  isNot: boolean;
+  promise: 'rejects' | 'resolves' | '';
+  utils: JestMatcherUtils;
+};
+
+type MatcherReturnType = {
+  message: () => string;
+  pass: boolean;
+  locator?: Locator;
+  name?: string;
+  expected?: unknown;
+  actual?: any;
+  log?: string[];
+};
+
+type MakeMatchers<R, T, ExtendedMatchers> = {
   /**
    * If you know how to test something, `.not` lets you test its opposite.
    */
-  not: MakeMatchers<R, T>;
+  not: MakeMatchers<R, T, ExtendedMatchers>;
   /**
    * Use resolves to unwrap the value of a fulfilled promise so any other
    * matcher can be chained. If the promise is rejected the assertion fails.
    */
-  resolves: MakeMatchers<Promise<R>, Awaited<T>>;
+  resolves: MakeMatchers<Promise<R>, Awaited<T>, ExtendedMatchers>;
   /**
    * Unwraps the reason of a rejected promise so any other matcher can be chained.
    * If the promise is fulfilled the assertion fails.
    */
-  rejects: MakeMatchers<Promise<R>, any>;
-} & IfAny<T, AllMatchers<R, T>, SpecificMatchers<R, T>>;
+  rejects: MakeMatchers<Promise<R>, any, ExtendedMatchers>;
+} & IfAny<T, AllMatchers<R, T>, SpecificMatchers<R, T> & ToUserMatcherObject<ExtendedMatchers, T>>;
 
-export type Expect = {
-  <T = unknown>(actual: T, messageOrOptions?: string | { message?: string }): MakeMatchers<void, T>;
-  soft: <T = unknown>(actual: T, messageOrOptions?: string | { message?: string }) => MakeMatchers<void, T>;
+export type Expect<ExtendedMatchers> = {
+  <T = unknown>(actual: T, messageOrOptions?: string | { message?: string }): MakeMatchers<void, T, ExtendedMatchers>;
+  soft: <T = unknown>(actual: T, messageOrOptions?: string | { message?: string }) => MakeMatchers<void, T, ExtendedMatchers>;
   poll: <T = unknown>(actual: () => T | Promise<T>, messageOrOptions?: string | { message?: string, timeout?: number, intervals?: number[] }) => BaseMatchers<Promise<void>, T> & {
     /**
      * If you know how to test something, `.not` lets you test its opposite.
      */
      not: BaseMatchers<Promise<void>, T>;
   };
-  extend(matchers: any): void;
+  extend<MoreMatchers extends Record<string, (this: State, receiver: unknown, ...args: any[]) => MatcherReturnType | Promise<MatcherReturnType>>>(matchers: MoreMatchers): Expect<ExtendedMatchers & MoreMatchers>;
```

Comment:
Judging by our own `toHaveLoggedSoftwareDownload`, forcing the return type is a breaking change.

Reply:
I think toHaveLoggedSoftwareDownload is not right, i'll fix it.